docs/DeveloperGuide.md

```diff
@@ -1,30 +1,126 @@
+insert SS of release page
+
+Click on `PlanPal.jar` to automatically download the file. Place the file in a folder your choice.
+
+
+
+## Design
+
+---
+### Architecture
+![Architecture.drawio.png](Images%2FArchitecture.drawio.png)
+
+The **Architecture Diagram** given above explains the high-level design of the program. Given below is a quick overview of the main components.
+
+### Main Components
+- `PlanPal`: Main entry of the program, initialises and connects the components
+- `UI`: In charge of printing messages
+- `Logic`: Determines the command to execute
+- `Storage`: Read and write data from hard disk
+- `Command`: Specific commands for execution
+
+### Program Flow
+- The `PlanPal` controls access to the `UI`, `Parser`, and `FileManager` (Storage) components of the app.
+- The user will send input to `PlanPal`.
+- `PlanPal` will parse the input to produce a `Command`, which will hook back into `PlanPal` to gain access to the other components when running
+
+---
+## Implementation
+---
+
+The class diagram below represents the contact book system
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+![ContactClassDiagram.jpg](Images/ContactClassDiagram.jpg)
 
+Currently, only the contact features have been illustrated.
+{to add other implementations for expense and activity managers}
 
+## Contact Manager Features
+
+### Add Command
+The sequence diagram below illustrates the process for resolving the "add" command.
+![AddContact.drawio.png](Images%2FAddContact.drawio.png)
+
+Explanation: 
+- Before this command is executed, the user will have to choose their mode.
+- When modeInput is 1, representing contact manager, the user is then asked for a command in this mode.
+- When command "add /name: john" is sent, the processCommand function is executed.
+- Since the command is "add", addContact function is executed using the description "/name: john". 
+- This creates a new contact and is then added to the list of contacts
+- Finally, contact file is saved in the savedContacts FileManager.
+
+### List Command
+The sequence diagram below illustrates the process for resolving the "list" command.
+
+![ViewContact.drawio.png](Images%2FViewContact.drawio.png)
+
+Explanation:
+- Before this command is executed, the user will have to choose their mode.
+- When modeInput is 1, representing contact manager, the user is then asked for a command in this mode.
+- When command "list" is sent, the processCommand function is executed.
+- Since the command is "add", viewContactList command is executed, printing all the contacts in the list.
+
+### Edit Command
+The sequence diagram below illustrates the process for resolving the "edit" command.
+![EditContact.drawio.png](Images%2FEditContact.drawio.png)
+
+### Set Category Command
+The sequence diagram below illustrates the process for resolving the "category" command.
+![SetCategory.drawio.png](Images%2FSetCategory.drawio.png)
```

Comment:
Should there be visualization of user just like the add command?
Set Category:
![image](https://github.com/user-attachments/assets/4bf05c03-07e3-42bb-87aa-5052897e943d)

Add: 
![image](https://github.com/user-attachments/assets/c123fc3f-5811-4b13-80ad-143e729b0afe)